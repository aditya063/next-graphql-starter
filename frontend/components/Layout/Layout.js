import PropTypes from 'prop-types';
import Header from '../Header/Header';
import Nav from '../Nav/Nav';
import Footer from '../Footer/Footer';

const Layout = React.memo(props => (
  <>
    <Header>
      <Nav />
    </Header>
    <main>
      {/* Pages are rendered here */}
      {props.children}
    </main>
    <Footer />
  </>
));

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default Layout;
