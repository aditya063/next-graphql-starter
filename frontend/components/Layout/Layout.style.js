import styled from 'styled-components';

export const divLayout = styled.div`
  display: grid;
  grid-template-areas:
    'header'
    'main';
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;

  @media screen and (min-width: 900px) {
    grid-template-areas:
      'header-margin-left header header-margin-right'
      '. main .';
    grid-template-columns: 1fr 900px 1fr;
  }
`;

export const headerMargin = styled.div`
  background-color: ${props => props.theme.color.black};
  display: none;

  @media screen and (min-width: 900px) {
    display: block;
  }
`;

export const headerMarginLeft = styled(headerMargin)`
  grid-area: header-margin-left;
`;

export const headerMarginRight = styled(headerMargin)`
  grid-area: header-margin-right;
`;

export const mainContainer = styled.main`
  /* display: flex; */
  /* flex-direction: column; */
  /* flex-grow: 1; */
  grid-area: main;
  /* max-width: 600; */
  /* margin: 0 auto; */
  /* width: 100%; */
`;
