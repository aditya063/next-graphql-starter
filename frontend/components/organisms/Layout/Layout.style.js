import styled from 'styled-components';
import { Layout } from 'antd';

//! Mobile first

export const AntLayout = styled(Layout)`
  /* display: flex; */
  /* flex-direction: column; */
  /* min-height: 100vh; */
`;

export const AntHeader = styled(Layout.Header)`
  padding: 0;

  /* h1 { */
  /* font-family: 'Play', sans-serif; */
  /* letter-spacing: 0.05rem; */
  /* } */
`;

export const AntContent = styled(Layout.Content)`
  padding: 20px;
`;

export const AntFooter = styled(Layout.Footer)``;

export const Container = styled.div`
  /* display: flex; */
  /* justify-content: center; */

  @media screen and (min-width: 900px) {
    /* margin: auto; */
    /* max-width: 900px; */
  }
`;
