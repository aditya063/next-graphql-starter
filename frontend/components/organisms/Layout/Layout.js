import PropTypes from 'prop-types';
import Footer from '../../molecules/Footer/Footer';
import * as sc from './Layout.style';

const Layout = props => (
  <sc.Wrapper>
    <sc.Header>
      <sc.Container>{props.header}</sc.Container>
    </sc.Header>

    <sc.Main>
      <sc.Container>{props.content}</sc.Container>
    </sc.Main>

    <sc.Footer>
      <sc.Container>
        <Footer />
      </sc.Container>
    </sc.Footer>
  </sc.Wrapper>
);

Layout.propTypes = {
  header: PropTypes.element,
  content: PropTypes.element
};

export default React.memo(Layout);
