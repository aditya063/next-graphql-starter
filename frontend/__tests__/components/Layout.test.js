import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../components/Layout/Layout';

describe('Layout', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    mockProps = { children: [] };
    wrapper = shallow(<Layout {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
