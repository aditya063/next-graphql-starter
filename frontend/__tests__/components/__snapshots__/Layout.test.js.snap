// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<Layoutstyle__divLayout>
  <Layoutstyle__headerMarginLeft />
  <Memo() />
  <Layoutstyle__headerMarginRight />
  <div
    key="1"
  >
    mock child component
  </div>
</Layoutstyle__divLayout>
`;
